src/tests/ftest/util/agent_utils.py

```diff
@@ -20,279 +20,158 @@
+        self.logfile = FormattedParameter("--logfile={}")
+
+
+class DaosAgentManager(SubprocessManager):
+    """Manages the daos_agent execution on one or more hosts using orterun."""
+
+    def __init__(self, ompi_path, agent_command):
+        """Initialize an DaosAgentManager object.
 
+        Args:
+            ompi_path (str): path to location of orterun binary.
+            agent_command (AgentCommand): server command object
+        """
+        super(DaosAgentManager, self).__init__(
+            "/run/agent_config", agent_command, ompi_path)
 
-def stop_agent(sessions, client_list=None):
-    """Kill ssh and the agent.
+    def stop(self):
+        """Stop the agent through the runner.
 
-    This is temporary; presuming the agent will deamonize at somepoint and can
-    be started and killed more appropriately.
+        Raises:
+            CommandFailure: if there was an errror stopping the agents.
 
-    Args:
-        sessions (dict): set of subprocess sessions returned by run_agent()
-        client_list (list, optional): lists of hosts running the daos agent.
-            Defaults to None.
+        """
+        self.log.info("Stopping agent orterun command")
 
-    Raises:
-        AgentFailed: if the daos agents failed to stop
+        # Maintain a running list of errors detected trying to stop
+        messages = []
 
-    """
-    # if empty client list, 'self' is effectively client
-    client_list = include_local_host(client_list)
-
-    # Kill the agents processes
-    pcmd(client_list, "pkill daos_agent", False)
-
-    # Kill any processes running in the sessions
-    for client in sessions:
-        if sessions[client].poll() is None:
-            sessions[client].kill()
-        sessions[client].wait()
-
-    # Check to make sure all the daos agents are dead
-    # pgrep exit status:
-    #   0 - One or more processes matched the criteria.
-    #   1 - No processes matched.
-    #   2 - Syntax error in the command line.
-    #   3 - Fatal error: out of memory etc.
-    time.sleep(5)
-    result = pcmd(client_list, "pgrep 'daos_agent'", False, expect_rc=1)
-    if len(result) > 1 or 1 not in result:
-        raise AgentFailed(
-            "DAOS agent processes detected after attempted stop on {}".format(
-                ", ".join([str(result[key]) for key in result if key != 1])))
+        # Stop the subprocess running the orterun command
+        try:
+            super(DaosAgentManager, self).stop()
+        except CommandFailure as error:
+            messages.append(
+                "Error stopping the orterun subprocess: {}".format(error))
+
+        # Kill any leftover processes that may not have been stopped correctly
+        self.kill()
+
+        # Report any errors after all stop actions have been attempted
+        if len(messages) > 0:
+            raise CommandFailure(
+                "Failed to stop agents:\n  {}".format("\n  ".join(messages)))
```

Comment:
(pylint-missing-final-newline) Final newline missing
(pylint-mixed-line-endings) Mixed line endings LF and CRLF